Update OpenVINO GenAI guide (#25283)
openvinotoolkit/openvino: docs/articles_en/learn-openvino/llm_inference_guide/genai-guide.rst

            std::string model_path = argv[1];
            ov::genai::LLMPipeline pipe(model_path, "CPU");

            auto streamer = [](std::string word) { 
               std::cout << word << std::flush; 

>                // Return flag correspods whether generation should be stopped.

Reviewer: ```suggestion
               // Return flag indicating whether generation should be stopped.
```

               // false means continue generation.
               return false;
            };
            pipe.generate("The Sun is yellow because", ov::genai::streamer(streamer));
         }

You can also create your custom streamer for more sophisticated processing:

.. tab-set::

   .. tab-item:: Python
      :sync: py

      .. code-block:: python

         import openvino_genai as ov_genai

         class CustomStreamer(ov_genai.StreamerBase):
            def __init__(self, tokenizer):
               ov_genai.StreamerBase.__init__(self)
               self.tokenizer = tokenizer
            def put(self, token_id) -> bool:
               # Decode tokens and process them.

>                # Streamer returns a flag correspoding whether generation should be stopped.

Reviewer: ```suggestion
               # Streamer returns a flag corresponding whether generation should be stopped.
```

>                # But in Python return can be ommited in that case the function will return None

Reviewer: ```suggestion
               # In Python `return` can be omitted. In that case the function will return None
```

>                # Streamer returns a flag corresponding whether generation should be stopped.
>                # In Python `return` can be omitted. In that case the function will return None
>                # which in turn will be converted to False, which means continue generation.

Reviewer 2: ```suggestion
               # Streamer returns a flag indicating whether generation should be stopped.
               # In Python, `return` can be omitted. In that case, the function will return None
               # which will be converted to False, meaning that generation should continue.
```

         class CustomStreamer: publict StreamerBase {
         public:
            bool put(int64_t token) {
               bool stop_flag = false; 
               /* 
               custom decoding/tokens processing code
               tokens_cache.push_back(token);
               std::string text = m_tokenizer.decode(tokens_cache);
               ...
               */

>                return stop_flag;  // flag whether generation should be stoped, if true generation stops.

Reviewer 2: ```suggestion
               return stop_flag;  // Flag indicating whether generation should be stopped. If True, generation stops.
```

            };

            void end() {
               /* custom finalization */
            };
         };

         int main(int argc, char* argv[]) {
            auto custom_streamer = std::make_shared<CustomStreamer>();

            std::string model_path = argv[1];
            ov::genai::LLMPipeline pipe(model_path, "CPU");
            pipe.generate("The Sun is yellow because", ov::genai::streamer(custom_streamer));
         }

> Using Generate in Chat Scenario

Reviewer 2: ```suggestion
Using GenAI in Chat Scenario
```
